Userland/Libraries/LibGfx/ImageFormats/PNGWriter.cpp

```diff
@@ -281,4 +322,54 @@ ErrorOr<ByteBuffer> PNGWriter::encode(Gfx::Bitmap const& bitmap, Options options
     return ByteBuffer::copy(writer.m_data);
 }
 
+class PNGAnimationWriter : public AnimationWriter {
+public:
+    PNGAnimationWriter(SeekableStream& stream)
+        : m_stream(stream)
+    {
+    }
+
+    virtual ErrorOr<void> add_frame(Bitmap&, int, IntPoint) override;
+
+private:
+    PNGWriter& m_writer;
+
+    bool is_first_frame { true };
+    u32 sequence_number { 0 };
+};
+
+ErrorOr<void> PNGAnimationWriter::add_frame(Bitmap& bitmap, int duration_ms, IntPoint at = {})
+{
+    // FIXME: Consider frame's duration and position
+    (void)at;
+
+    // Let's get rid of the previously written trailer
+    if (!is_first_frame)
+        TRY(m_stream.seek(-4, SeekMode::FromEndPosition));
```

Comment:
While I still have that in mind:
Please use `SeekMode::FromCurrentPosition` (https://github.com/SerenityOS/serenity/pull/24412/commits/89aa18f2b89f99eb1f0ed58d8fa08b8137443a09)

Reply:
Cheers! Will get on that asap!